src/coreclr/jit/lsraarm64.cpp

```diff
@@ -2059,7 +2036,7 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                 {
                     SingleTypeRegSet candidates = lowVectorOperandNum == 2 ? lowVectorCandidates : RBM_NONE;
 
-                    if (intrin.op2->gtType == TYP_MASK)
+                    if (intrin.op2->OperIsHWIntrinsic(NI_Sve_ConvertVectorToMask))
```

Comment:
Why does this change the register set to MASK but regular TYP_MASK nodes do not?

Reply:
Its the same reason I mentioned in https://github.com/dotnet/runtime/pull/102611#discussion_r1645030286